Description: [PATCH] use zend_ce_exception instead of zend_exception_get_default() for 8.5
Author: Remi Collet <remi@remirepo.net>
Origin: upstream, https://github.com/nono303/php-geos/commit/cd8fed75d9b0893ce9ac97672b747eddf991ea57
Last-Update: 2026-01-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/geos.c b/geos.c
index ad2dce2..8f4bdd3 100644
--- a/geos.c
+++ b/geos.c
@@ -150,7 +150,11 @@ static void errorHandler(const char *fmt, ...)
     va_end(args);
 
     /* TODO: use a GEOSException ? */
+#if PHP_VERSION_ID < 80000
     zend_throw_exception_ex(zend_exception_get_default(TSRMLS_C),
+#else
+    zend_throw_exception_ex(zend_ce_exception,
+#endif
         1 TSRMLS_CC, "%s", message);
 
 }
